algorithms/src/snark/marlin/ahp/ahp.rs

```diff
@@ -185,17 +149,22 @@ impl<F: PrimeField, MM: MarlinMode> AHPForR1CS<F, MM> {
         let largest_non_zero_domain =
             Self::max_non_zero_domain_helper(state.non_zero_a_domain, state.non_zero_b_domain, state.non_zero_c_domain);
 
-        let public_input = prover::ConstraintSystem::format_public_input(public_input);
-        if !Self::formatted_public_input_is_admissible(&public_input) {
-            return Err(AHPError::InvalidPublicInputLength);
+        let public_input =
+            public_input.iter().map(|p| prover::ConstraintSystem::format_public_input(p)).collect::<Vec<_>>();
+
+        for public_input in &public_input {
+            if !Self::formatted_public_input_is_admissible(public_input) {
+                return Err(AHPError::InvalidPublicInputLength);
+            }
         }
-        let input_domain = EvaluationDomain::new(public_input.len()).ok_or(AHPError::PolynomialDegreeTooLarge)?;
+        let input_domain = EvaluationDomain::new(public_input[0].len()).ok_or(AHPError::PolynomialDegreeTooLarge)?;
```

Comment:
we might want to guard against empty `public_input` before this call, in order to not potentially panic on indexing